### PR TITLE
Allow for higher resolution return from pressure sensor

### PR DESCRIPTION
**Pull request by dwblair**
Hi All!

Thanks so much for your awesome work on this sensor -- I'm finding it super useful.

I believe that the reason that the library was returning only integer values of the pressure value involved integer vs. floating point division.

I've made some small changes to the calculation of both P and the pressure, so that the returned value has a higher resolution.

I'm new to doing pull requests, figured I'd give it a shot :)

Cheerio!
D

### CODE REVIEW

File: MS5837.cpp [hunk 1]
```diff
@@ -150,17 +150,22 @@ void MS5837::calculate() {
...
 	} else {
-		TEMP = (TEMP-Ti);
-		P = (((D1*SENS2)/2097152l-OFF2)/8192l)/10;
+		P = (((D1*SENS2)/2097152l-OFF2)/8192l);
 	}
 }
 
 float MS5837::pressure(float conversion) {
-	return P*conversion;
+    if ( _model == MS5837_02BA ) {
```
[the remark this answers is not in the document]
Comment (dwblair) [reply MS5837.cpp]: Ah, cool!  That was my original thought, too -- but the variable P is defined as int32_t in MS5837.h, so I don't think I can store the result of a floating point division in it ... 

Would it be more elegant for me to redefine P as a 'float' in MS5837.h, and then make the change you're suggesting?

Apologies, not an expert coder :)

**Comment by woessmich** [comment]
Hi
The change to higher resolution does not work for me like that. In Arduino IDE, the calculation needs brackets so that first the multiplication is done, afterwards the division by 10 or 100. Otherwise we end up with the same result and no better resolution.
The section should look like:

```
float MS5837::pressure(float conversion) {
    if ( _model == MS5837_02BA ) {
        return (P*conversion)/100.0f;
    }
    else {
        return (P*conversion)/10.0f;
    }
}

File: MS5837.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaxxzer) [reply MS5837.cpp]: On second thought, the suggestion might lead to more intermediary float calculations. This patch is fine.

**Comment by jaxxzer** [comment]
This fixes #7 !

**Comment by jaxxzer** [comment]
@woessmich Can you verify your problem? It is working fine here.

**Pull request merged**